src/gossip.rs

```diff
@@ -119,8 +133,12 @@ pub async fn get_gossip_sync(
 
     // if we error out, we just use the default gossip data
     let gossip_data = match get_gossip_data(&rexie, logger.clone()).await {
-        Ok(gossip_data) => gossip_data,
-        Err(_) => Gossip::new(network, logger.clone()),
+        Ok(Some(gossip_data)) => gossip_data,
+        Ok(None) => Gossip::new(network, logger.clone()),
+        Err(e) => {
+            error!("Error getting gossip data from storage: {e}, re-syncing gossip...");
+            Gossip::new(network, logger.clone())
+        }
```

Comment:
You could use `unwrap_or_else` twice to get rid of the multiple Ok/Some/None parts. 

```
let gossip_data = get_gossip_data(&rexie, logger.clone())
    .await
    .unwrap_or_else(|e| {
        error!("Error getting gossip data from storage: {e}, re-syncing gossip...");
        None
    })
    .unwrap_or_else(|| Gossip::new(network, logger.clone()));
```

Reply:
yeah I considered that but I think the match is cleaner and more clear on what is happening in each case